CoreGraphUtilities/src/au/gov/asd/tac/constellation/graph/utilities/hashmod/HashmodAction.java

```diff
@@ -133,16 +129,23 @@ private static void run(final GraphWriteMethods wg, final PluginInteraction inte
                     newAttributeType = attributeName[attributeName.length - 1];
                 }
 
-                if (createAttributes) {
-                    final int newAttribute = wg.addAttribute(GraphElementType.VERTEX, newAttributeType, nextAttr, nextAttr, "", null);
-                    if (newAttribute != Graph.NOT_FOUND) {
+                if (createVertices || createAttributes) {
+                    //check if it's an existing attribute
+                    int existingAttribute = wg.getAttribute(GraphElementType.VERTEX, nextAttr);
+
+                    // do not add new attribute if not createAttributes and
+                    // does not exist in graph
+                    if (!createAttributes && existingAttribute < 0) {
```

Comment:
instead of `existingAttribute < 0`, can you make this `existingAttribute == Graph.NOT_FOUND` as that's the value that will be returned if the attribute doens't exist